### PR TITLE
Force backend by WGPU_BACKEND_TYPE env var

### PR DESCRIPTION
**Pull request by berendkleinhaneveld**
This makes sure that on other platforms than Windows any available backend is picked.

### CODE REVIEW

File: wgpu/backends/rs.py [hunk 1]
```diff
@@ -217,7 +218,9 @@ def request_adapter(self, *, canvas, power_preference=None):
         # mature than Metal too, so let's just force that for now.
```
Comment (Korijn) [inline wgpu/backends/rs.py:218]: Can you update the comment accordingly?

**Comment by Korijn** [comment]
Is the Metal backend mature enough for usage yet?

**Comment by almarklein** [comment]
> Is the Metal backend mature enough for usage yet?

I guess Berend will find out :)

**Comment by Korijn** [comment]
As discussed: let's only force a backend if a certain envvar is given

File: wgpu/backends/rs.py [hunk 2]
```diff
@@ -213,14 +213,35 @@ def request_adapter(self, *, canvas, power_preference=None):
...
+        # the NVidia control panel settings.
         # See https://github.com/gfx-rs/wgpu/issues/1416
-        # force_backend = lib.WGPUBackendType_Vulkan
-        force_backend = enum_str2int["BackendType"][
-            "Vulkan" if sys.platform.startswith("win") else "Null"
-        ]
+        force_backend = enum_str2int["BackendType"]["Null"]
+        if "WGPU_BACKEND_TYPE" in os.environ:
+            try:
+                backend = int(os.environ["WGPU_BACKEND_TYPE"])
```
Comment (Korijn) [inline wgpu/backends/rs.py:226]: do you have to pick by integer? that seems... unfriendly?

Comment (berendkleinhaneveld) [reply wgpu/backends/rs.py:226]: Good call!

**Comment by Korijn** [comment]
LGTM!

**Comment by almarklein** [comment]
I would expect this to fail, but I forgot that I had simply disabled all render-tests on CI. With this change, there are some tests running on CI (e.g. juggling with buffers and textures) which were previously skipped :)

**Pull request merged**